packages/file-search/src/node/file-search-service-impl.spec.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import 'mocha';
+import * as chai from 'chai';
+import * as path from 'path';
+import { FileSearchServiceImpl } from './file-search-service-impl';
+import { FileUri } from '@theia/core/lib/node';
+import { MockLogger } from '@theia/core/lib/common/test/mock-logger';
+
+const expect = chai.expect;
+
+describe('search-service', () => {
+    it('shall fuzzy search this spec file', async () => {
+        const service = new FileSearchServiceImpl(new MockLogger());
+        const uri = FileUri.create(path.resolve(__dirname, ".."));
+        const matches = await service.find(uri.toString(), 'spc');
```

Comment:
The problem is here on Windows, you will get back a couple of URI strings which are:
```
matches [ 'file:///c%3A/Users/kittaakos/dev/theia/packages/file-search/src/browser/file-search-frontend-module.ts',
  'file:///c%3A/Users/kittaakos/dev/theia/packages/file-search/src/browser/quick-file-open-contribution.ts',
  'file:///c%3A/Users/kittaakos/dev/theia/packages/file-search/src/browser/quick-file-open.ts',
  'file:///c%3A/Users/kittaakos/dev/theia/packages/file-search/src/common/file-search-service.ts',
  'file:///c%3A/Users/kittaakos/dev/theia/packages/file-search/src/node/file-search-backend-module.ts',
  'file:///c%3A/Users/kittaakos/dev/theia/packages/file-search/src/node/file-search-service-impl.spec.ts',
  'file:///c%3A/Users/kittaakos/dev/theia/packages/file-search/src/node/file-search-service-impl.ts',
  'file:///c%3A/Users/kittaakos/dev/theia/packages/file-search/src/node/git-ignore.spec.ts',
  'file:///c%3A/Users/kittaakos/dev/theia/packages/file-search/src/node/git-ignore.ts',
  'file:///c%3A/Users/kittaakos/dev/theia/packages/file-search/src/node/gitignore-parser.d.ts' ]
```

And your file is:
```
C:\Users\kittaakos\dev\theia\packages\file-search\src\node\file-search-service-impl.spec.ts
```

So conceptually, we usually ignore, that the URI string is always encoded in Windows, that is usually not the case on OS X.

Long story short: you cannot just write `endsWith`. That was a problem is Git too. I hope this helps.

Reply:
This should fix the test: `const testFile = matches.find(e => e.endsWith(FileUri.create(__filename).toString()));`

Reply:
thanks!